azkaban-common/src/main/java/azkaban/executor/ExecutorHealthChecker.java

```diff
@@ -169,11 +179,32 @@ private void handleExecutorNotAliveCase(
         (executor.getId(), 0) + 1);
     if (this.executorFailureCount.get(executor.getId()) % this.executorMaxFailureCount == 0
         && !this.alertEmails.isEmpty()) {
-      entry.getValue().stream().forEach(flow -> flow
-          .getExecutionOptions().setFailureEmails(this.alertEmails));
       logger.info(String.format("Executor failure count is %d. Sending alert emails to %s.",
           this.executorFailureCount.get(executor.getId()), this.alertEmails));
-      this.alerterHolder.get("email").alertOnFailedUpdate(executor, entry.getValue(), e);
+      this.alerterHolder.get("email").alertOnFailedExecutorHealthCheck(executor, entry.getValue(), e,
+          this.alertEmails);
+      this.cleanupForMissingExecutor(executor, entry.getValue());
```

Comment:
Should we make auto-remediation optional through a configuration variable?

Reply:
I thought about having a config for this originally but started wondering if there would be a good reason for someone to not attempt remediation (mainly the removal of the executor from db).
Configs like these are also have the side effect either resulting a in large number of tests that need to be maintained for different config values or inadequate testing for some config value(s).

I am open to the idea however. Is there any notion of `experimental` configs followed in Azkaban where a new feature is introduced with a config but after a sufficient 'soak time' (in production) the configuration is removed ?

Reply:
I have had switch-controlled-feature-introduction on my mind. It requires two steps. First, there is an on-off switch/flag/config for a feature we introduce. Later, in step 2, after sufficient soaking the flag is removed. The drawback is the second step of removing the flag is a manual step, and requires code change as well. 

What approach did you have in mind?

Reply:
I had a similar idea in mind but with a few caveats.
(1) Configs introduced in this manner should have something within the config name to indicate that it will be removed in the near future and the only supported  behavior at that point will then be whatever is added as the `default` for the config
(2) This should still be done selectively for features of some magnitude, as it
- Puts an additional burden on someone to remove the config later, by then developers might have lost some context of the change and would make it even harder to debug any issues at that point
- Could result in a combinatorial explosion of tests that need to be maintained or some of the config values will remain untested if the config ends up staying long term